Contributing guidelines [8715] (#167)
eProsima/Fast-RTPS-docs: CONTRIBUTING.md

# Contribution Guidelines

*Fast DDS-Docs* is an open source project for documenting the *Fast DDS* library and some related tools such as *Fast DDS-Gen*.

> In that sense, contributions to expand or improve the documentation are most welcome and encouraged; we just want the best documentation possible for out community.

Reviewer: ```suggestion
In that sense, contributions to expand or improve the documentation are most welcome and encouraged; we just want the best documentation possible for our community.
```

There are two ways to contribute to this project:

* [Opening an issue](#opening-an-issue).
* [Submitting a pull request](#submitting-a-pull-request).

Furthermore, this document also provide the contributor with a [Project structure](#project-structure) definition, and with a [Style guide](#style-guide) to ease the writing process.

## Opening an issue

The easiest way for you to point out an incorrect or missing piece of documentation is to open an issue in the GitHub repository.
When doing so, please take the following into account:

1. Try to use positive and welcoming language.
We sincerely appreciate your contribution and want to help each other.
1. Describe your needs or concerns in detail.

> The more information provided, the better our understanding there will be, and the more satisfied you will get in the end.

Reviewer: ```suggestion
The more information provided, the better our understanding will be, and the more satisfied you will get in the end.
```

1. Be patient.
Bear in mind that your issue might not be answered or addressed right away.
If a couple of days go by without any feedback, please send a friendly ping by commenting on the issue.
1. Provide feedback.
Once your issue is addressed, please provide feedback so we can improve the process as we go.

## Submitting a pull request

If writing an issue does not do it for you because you just know what has to be done and how, and you have the energy and time to directly contribute to the documentation, submitting a pull request is your way to go.

> Please mind that in doing so, you accept to abide to the pull request process followed on this repository, meaning:

Reviewer: ```suggestion
Please mind that by doing so, you accept to abide to the pull request process followed on this repository, meaning:
```

1. Stick to the [Project structure](#project-structure) and [Style guide](#style-guide).
1. Test your changes before submission, and only submit "green" pull requests.
Our Continuous Integration will still run the tests, but this way the process becomes smoother and the "Fixing errors" commits and re-runs are avoided.
Follow the setup instructions outlined in [README.md](README.md) to prepare your development environment.
If you need any help setting up, please let us know by opening an issue.
We will be happy to help you and to improve this repository's [README.md](README.md) instructions.
1. Be prepared to received criticism, and potentially tons of change requests.
We want our documentation to be as complete, concise, precise, and helpful as possible, so we put a lot of focus on the details.
Please do not take criticism as personal, we really appreciate your contributing effort.
1. Mind that [LICENSE](LICENSE) applies to all contributions made to this repository.

## Project structure

> The project follows the following structure:

Reviewer: The "follows the following" sound strange, I think:

```suggestion
The project is structured as follows:
```

1. The root directory contains global scope files, such as this one.
1. The [docs directory](#docs-directory) contains all documentation source code.
1. Code snippets and testing code is located in the [code directory](#code-directory).

> ### docs directory

Reviewer: ```suggestion
### `docs` directory
```

The [docs](docs) directory contains:

* [_static](docs/_static): For HTML theme related files.
* [01-figures](docs/01-figures): For all the documentation figures. SVG files are the preferred format since the XML can be modified or otherwise checked for differences.
* [02-formalia](docs/02-formalia): Must-have pages such as Introduction.
* [fastdds](docs/fastdds): Fast DDS documentation.
* [fastddsgen](docs/fastddsgen): Fast DDS-Gen documentation.
* [installation](docs/installation): Installation manual.
* [notes](docs/notes): Release notes.

All new documentation must fall into one of these directories, with the exception of those contributions which are not related to any of the given descriptions.
Keep in mind that this is an Sphinx based project, and as such, the all the documentation is written in `reStructuredText`.

All unrecognized words must be added to the [spelling_wordlist.txt](docs/spelling_wordlist.txt) dictionary in alphabetical order, with exception of the ones coming from the API reference documentation, which must be added to [docs/fastdds/api_reference/spelling_wordlist.txt](docs/fastdds/api_reference/spelling_wordlist.txt).

> ### code directory

Reviewer: ```suggestion
### `code` directory
```

1. In this sense, avoid contractions of any kind.
1. Direct speech may be used on tutorials, but it should be replaced with imperative mode with no subject when possible.
1. Always state what an acronym stands for the first time you use it.
Try to also state it from time to time as you go.

### Text structure

1. Go from the abstract to the concrete.
Give a big picture first and then start digging into the specifics.
1. Always write keeping maintainability in mind.
1. Split you content in different pages as much as readability allows.
This makes it easier for everyone to find just the piece of information they need without scrolling back and forth over endlessly long pages.
1. Tables are a great form of comprising a lot of data in a small, easy to consult manner.
Use them as much as you can, even if it duplicates some information.
1. Use as many `note`, `important`, and `warning` blocks as you need.
They are a great way to convey a really important message the reader should not overlook.
1. Add cross-references to other relevant sections, API reference, and external sites or documents as much as possible.

> Make sure the reader has easy access to all relevant information they may need for fully comprehending the documentation.

Reviewer: In this context I think it would be *need + to-infinitive*.

```suggestion
Make sure the reader has easy access to all relevant information they may need to fully comprehend the documentation.
```

Author: Both work IMHO. I'll take yours
